### PR TITLE
Docs/issue 1661 add tip to source docs and update weaviate docs

### PR DESCRIPTION
**Pull request by dat-a-man**
### Description

#1661 

This PR clarifies the behavior of the `max_table_nesting` parameter, ensuring users understand that it doesn’t automatically apply to individual resources. It provides guidance on how to correctly apply the parameter to achieve the desired effect.

### CODE REVIEW

File: docs/website/docs/dlt-ecosystem/destinations/weaviate.md [hunk 1]
```diff
@@ -116,6 +116,22 @@ weaviate_adapter(
     tokenization={"title": "word", "description": "whitespace"},
 )
 ```
+When adapting resources, it's important to implement changes directly to the data source itself rather than extracting resources separately. Here's how:
```
Comment (VioletM) [inline docs/website/docs/dlt-ecosystem/destinations/weaviate.md:119]:
```suggestion
When using the `weaviate_adapter`, it's important to apply it directly to resources, not to the whole source. Here's an example:
```

File: docs/website/docs/dlt-ecosystem/destinations/weaviate.md [hunk 2]
```diff
@@ -116,6 +116,22 @@ weaviate_adapter(
...
+
+```py
+products_tables = sql_database().with_resources("products", "customers")
+
+pipeline = dlt.pipeline(
+        pipeline_name="postgres_to_weaviate_pipeline",
+        destination="weaviate",
+    )
+
+# adapt the resource within the source
```
Comment (VioletM) [inline docs/website/docs/dlt-ecosystem/destinations/weaviate.md:129]:
```suggestion
# apply adapter to the needed resources
```

File: docs/website/docs/general-usage/source.md [hunk 3]
```diff
@@ -187,6 +187,12 @@ Several data sources are prone to contain semi-structured documents with very de
 MongoDB databases. Our practical experience is that setting the `max_nesting_level` to 2 or 3
 produces the clearest and human-readable schemas.
 
+:::tip TIP
+The `max_table_nesting` parameter at the source level doesn't automatically apply to individual 
+resources when accessed directly (e.g., using `source.resources["resource_1"])`. To make sure it 
+works, either use `source.with_resources("resource_1")` or set the parameter directly on the resource.
```
Comment (VioletM) [inline docs/website/docs/general-usage/source.md:193]: please, add an example how parameter could be set directly in the resource

**Review by VioletM — CHANGES_REQUESTED** [review]
Looks really good, added some comments :)

**Comment by VioletM** [comment]
Also, please add the same info about adapter for qdrant https://deploy-preview-1662--dlt-hub-docs.netlify.app/docs/dlt-ecosystem/destinations/qdrant/#qdrant_adapter and lancedb https://deploy-preview-1662--dlt-hub-docs.netlify.app/docs/dlt-ecosystem/destinations/lancedb/#run-the-pipeline

File: docs/website/docs/dlt-ecosystem/destinations/weaviate.md (hunk 2 above)
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/destinations/weaviate.md:129]: done

File: docs/website/docs/dlt-ecosystem/destinations/weaviate.md (hunk 1 above)
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/destinations/weaviate.md:119]: done

File: docs/website/docs/general-usage/source.md (hunk 3 above)
Comment (dat-a-man) [reply docs/website/docs/general-usage/source.md:193]: done

File: docs/website/docs/dlt-ecosystem/destinations/qdrant.md [hunk 4]
```diff
@@ -106,10 +106,25 @@ qdrant_adapter(
...
 
-:::tip
+When using the `qdrant_adapter`, it's important to apply it directly to resources, not to the whole source. Here's an example:
 
-A more comprehensive pipeline would load data from some API or use one of dlt's [verified sources](../verified-sources/).
+```py
+products_tables = sql_database().with_resources("products", "customers")
+
+pipeline = dlt.pipeline(
+        pipeline_name="postgres_to_weaviate_pipeline",
```
Comment (VioletM) [inline docs/website/docs/dlt-ecosystem/destinations/qdrant.md:115]:
```suggestion
        pipeline_name="postgres_to_qdrant_pipeline",
```

File: docs/website/docs/dlt-ecosystem/destinations/qdrant.md [hunk 5]
```diff
@@ -106,10 +106,25 @@ qdrant_adapter(
...
+        destination="weaviate",
```
Comment (VioletM) [inline docs/website/docs/dlt-ecosystem/destinations/qdrant.md:116]:
```suggestion
        destination="qdrant",
```

File: docs/website/docs/general-usage/source.md [hunk 6]
```diff
@@ -187,6 +187,22 @@ Several data sources are prone to contain semi-structured documents with very de
...
+
+You can directly configure the `max_table_nesting` parameter on the resource level as:
+
+```py
+@dlt.resource(max_table_nesting=0)
+def my_resource():
+    ...
+```
+
+
```
Comment (VioletM) [inline docs/website/docs/general-usage/source.md:205]:
```suggestion
or 

```py
my_source = my_source()
my_source.my_resource.max_table_nesting = 0
```

```

**Review by VioletM — CHANGES_REQUESTED** [review]

File: docs/website/docs/dlt-ecosystem/destinations/lancedb.md [hunk 7]
```diff
@@ -144,7 +144,22 @@ lancedb_adapter(
...
 ```
 
-Bear in mind that you can't use an adapter on a [dlt source](../../general-usage/source.md), only a [dlt resource](../../general-usage/resource.md).
+When using the `lancedb_adapter`, it's important to apply it directly to resources, not to the whole source. Here's an example:
+
+```py
+products_tables = sql_database().with_resources("products", "customers")
+
+pipeline = dlt.pipeline(
+        pipeline_name="postgres_to_weaviate_pipeline",
```
[the remark this answers is not in the document]
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/destinations/lancedb.md:153]: done

File: docs/website/docs/dlt-ecosystem/destinations/lancedb.md [hunk 8]
```diff
@@ -144,7 +144,22 @@ lancedb_adapter(
...
+        destination="weaviate",
```
[the remark this answers is not in the document]
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/destinations/lancedb.md:154]: done

File: docs/website/docs/dlt-ecosystem/destinations/qdrant.md (hunk 4 above)
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/destinations/qdrant.md:115]: done

File: docs/website/docs/dlt-ecosystem/destinations/qdrant.md (hunk 5 above)
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/destinations/qdrant.md:116]: done

File: docs/website/docs/general-usage/source.md (hunk 6 above)
Comment (dat-a-man) [reply docs/website/docs/general-usage/source.md:205]: done

**Review by VioletM — APPROVED** [review]

**Pull request merged**